bin/node-bindings/odiff.d.ts

```diff
@@ -1,4 +1,8 @@
 export type ODiffOptions = Partial<{
+  /** The image type of the base image. This has to be set to the corresponding image format when using a buffer as input */
+  baseImageType?: 'filepath' | 'jpg' | 'png' | 'bmp' | 'tiff' = 'filepath';
+  /** The image type of the compare image. This has to be set to the corresponding image format when using a buffer as input */
+  compareImageType?: 'filepath' | 'jpg' | 'png' | 'bmp' | 'tiff' = 'filepath';
```

Comment:
I don't really like, that you are currently able to provide a buffer as input and at the same time add "filepath" as the image type.
Maybe this can be prevented with some typescript magic, but I am really not experienced in ts.

Reply:
Is there any way too quickly read codec info from buffer? I understand why this is done looks like it may hurt performance to add a lookup over all the codecs.

I think this can be done by simply duplicating overrides like this 

```ts
declare function compare(
  baseImage: Buffer,
  compareImage: buffer,
  baseCodec: "..",
  compareCodec: ".."
)

declare function compare(
  baseImage: string,
  compareImage: string,
)
``` 

and then in function check the type of baseImage && compareImage